### ISSUE TITLE
Move and change `transform_sigmoid` parameter to GraphDataset

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#91]
NeuralNet class has `transform_sigmoid` parameter, which in `reg` cases maps the **predicted output** to the domain [0, 1] using a sigmoid function. 

Talking with @DTRademaker and Daniil, I understood that what we want is a bit different though: 
- The ground truth target (BA) should be first transformed with a log function, and then mapped to [0, 1] domain using a sigmoid function. ** This possibility is not currently implemented, right @cbaakman? 
- The `transform_sigmoid` parameter, for as it is now, only modifies the predicted output (for optimization reasons, and NOT for the above mentioned reasons), and not the ground truth output. So I think it's not doing it right anyway, because doing so the ground truth values and the predicted ones will have a different domain (the former can have whatever domain, while the latter has [0, 1] domain). Consequently, any metrics would be wrong because we are not using values mapped to the same domain to compute it. Could you confirm that this is the case @cbaakman?
- I will write about how I think it would be best to implement this functionality (mapping the target to a particular domain) after I am sure I am headed in the right direction.

-----------------

\** _Exaplaination for this_: the data we have is mostly made of very strong binders, and the target distribution resembles a decreasing exponential. In order to avoid training a model that will learn well for the strongest binders and poorly for the weakest ones, log and sigmoid transformation should be applied to this data. The resulting distribution is then much more uniform, and thus easier for the model to predict. Additionally, now binary cross-entropy can be used as a loss function, thus evenly distributing the error for low values and big values (not weighted MSE gives greater errors for very low and smaller errors for very high BA values). Alternatives could have been 1) Oversampling the data such that the distribution of BAs is uniform instead of exponential, or 2) Using weighted MSE to account for small vs big values.

**Comment by gcroci2** [comment]
- The ground truth target (BA in our 3D-Vac application) should be first transformed with a log function, and then mapped to [0, 1] domain using a sigmoid function (not currently implemented).
- Add tests to confirm this mapping.
- Add tests to confirm the predicted output mapping.
- Improve docs in the code about it.
- Add example in the readme.
- Raise an error if the user tries to set `transform_sigmoid` as True but the task is `reg`.
